### PR TITLE
Keep Project.json in sync with TPMinV for UWP Projects

### PR DESCRIPTION
**Pull request by shishirx34**
Fixes https://github.com/NuGet/Home/issues/4932. 

Replaces the UAP framework in project.json with the TPMinV from csproj for UWP projects. 

Added relevant unit tests. Investigating into adding end to end tests if possible.

### CODE REVIEW

File: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/ProjectJsonBuildIntegratedProjectSystem.cs [hunk 1]
```diff
@@ -35,6 +36,23 @@ public class ProjectJsonBuildIntegratedProjectSystem : ProjectJsonBuildIntegrate
             var projectId = VsHierarchyUtility.GetProjectId(envDTEProject);
             InternalMetadata.Add(NuGetProjectMetadataKeys.ProjectId, projectId);
 
+            // Override the JSON TFM value from the csproj for UAP framework
+            if (InternalMetadata.TryGetValue(NuGetProjectMetadataKeys.TargetFramework, out object targetFramework))
```
Comment (alpaix) [inline src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/ProjectJsonBuildIntegratedProjectSystem.cs:40]: This won't compile for VS14.

File: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/ProjectJsonBuildIntegratedProjectSystem.cs [hunk 2]
```diff
@@ -35,6 +36,23 @@ public class ProjectJsonBuildIntegratedProjectSystem : ProjectJsonBuildIntegrate
...
+            {
+                var jsonTargetFramework = targetFramework as NuGetFramework;
+                if (IsUAPFramework(jsonTargetFramework))
+                {
+                    var platfromMinVersion = VsHierarchyUtility.GetMSBuildProperty(VsHierarchyUtility.ToVsHierarchy(envDTEProject), EnvDTEProjectInfoUtility.TargetPlatformMinVersion);
+
+                    if (platfromMinVersion != null)
```
Comment (alpaix) [inline src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/ProjectJsonBuildIntegratedProjectSystem.cs]: `!string.IsNullOrEmpty()`?
- addressed.

File: src/NuGet.Core/NuGet.PackageManagement/Projects/ProjectJsonBuildIntegratedNuGetProject.cs [hunk 3]
```diff
@@ -308,12 +318,26 @@ private JObject GetJson()
...
             {
                 await writer.WriteAsync(json.ToString());
             }
         }
+
+        private void UpdateFramework(JObject json)
+        {
+            var frameworks = JsonConfigUtility.GetFrameworks(json);
+            if (frameworks.Count() == 1
+                && IsUAPFramework(frameworks.First())
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.PackageManagement/Projects/ProjectJsonBuildIntegratedNuGetProject.cs]: I'd rather implement this method to fit generic scenario by simply comparing json framework against the metadata one.

File: src/NuGet.Core/NuGet.PackageManagement/Utility/JsonConfigUtility.cs [hunk 4]
```diff
@@ -173,6 +214,14 @@ private static JObject SortProperties(JObject parent)
             return sortedParent;
         }
 
+        private static bool HasFramework(IEnumerable<NuGetFramework> list, NuGetFramework framework)
+        {
+            return list.Any(
+                item =>
+                    string.Equals(item.Framework, framework.Framework, StringComparison.OrdinalIgnoreCase)
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.PackageManagement/Utility/JsonConfigUtility.cs]: `NuGetFramework` employs its own comparer, so `list.Any(framework)` should just work.

File: test/NuGet.Core.Tests/NuGet.PackageManagement.Test/ProjectManagement/JsonConfigUtilityTests.cs [hunk 5]
```diff
@@ -109,6 +109,59 @@ public void JsonConfigUtility_GetTargetFramework()
         }
 
         [Fact]
+        public void JsonConfigUtility_AddFramework()
+        {
+            // Arrange
+            var json = BasicConfig;
+
+            // Act
```
Comment (alpaix) [inline test/NuGet.Core.Tests/NuGet.PackageManagement.Test/ProjectManagement/JsonConfigUtilityTests.cs]: nit: `Act` normally corresponds to a single line calling a method the test is written for. In this case it's line 121.

File: src/NuGet.Core/NuGet.PackageManagement/Utility/JsonConfigUtility.cs [hunk 6]
```diff
@@ -157,6 +157,47 @@ public static IEnumerable<NuGetFramework> GetFrameworks(JObject json)
...
+            var frameworks = GetFrameworks(json);
+            if (HasFramework(frameworks, framework))
+            {
+                return;
+            }
+
+            JObject frameworkSet = null;
+
+            JToken node = null;
+            if (json.TryGetValue("frameworks", out node))
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.PackageManagement/Utility/JsonConfigUtility.cs]: Add a constant for "frameworks"
- addressed.

File: src/NuGet.Clients/NuGet.VisualStudio.Common/EnvDTEProjectInfoUtility.cs [hunk 7]
```diff
@@ -24,6 +24,12 @@ public static class EnvDTEProjectInfoUtility
 
         public const string WebConfig = "web.config";
         public const string AppConfig = "app.config";
+        public const string FullPath = "FullPath";
```
Comment (alpaix) [inline src/NuGet.Clients/NuGet.VisualStudio.Common/EnvDTEProjectInfoUtility.cs:27]: Don't have to be public, right?

**Review by alpaix — CHANGES_REQUESTED** [review]

File: src/NuGet.Clients/NuGet.VisualStudio.Common/EnvDTEProjectInfoUtility.cs (hunk 7 above)
Comment (shishirx34) [reply src/NuGet.Clients/NuGet.VisualStudio.Common/EnvDTEProjectInfoUtility.cs:27]: This doesn't need to be but TargetPlatformMinVersion does need to be to reuse above to get TPMin version from envDTEProject. Ideally I would have liked these values to be consts defined in a separate class or something, but I am avoiding to make any unneeded refactors for now.

File: src/NuGet.Core/NuGet.PackageManagement/Utility/JsonConfigUtility.cs (hunk 4 above)
Comment (shishirx34) [reply src/NuGet.Core/NuGet.PackageManagement/Utility/JsonConfigUtility.cs]: Oh cool, I wasn't aware it had a comparer

File: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/ProjectJsonBuildIntegratedProjectSystem.cs (hunk 1 above)
Comment (shishirx34) [reply src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/ProjectJsonBuildIntegratedProjectSystem.cs:40]: I ran the build script and saw no errors. How else can I compile for VS14?

File: src/NuGet.Core/NuGet.PackageManagement/Projects/ProjectJsonBuildIntegratedNuGetProject.cs (hunk 3 above)
Comment (shishirx34) [reply src/NuGet.Core/NuGet.PackageManagement/Projects/ProjectJsonBuildIntegratedNuGetProject.cs]: Had an offline chat with Alex, decided to add a check on the `newTargetFramework` for UAP and get input from PM on the need for check for json framework.

**Review by alpaix — APPROVED** [review]

**Pull request merged**